launchpod/src/test/java/com/google/launchpod/FormHandlerServletTest.java

```diff
@@ -0,0 +1,403 @@
+import org.mockito.MockitoAnnotations;
+
+/** 
+ * Runs unit tests for the FormHandlerServlet that contains doPost(), doGet(), and xmlString() methods.
+*/
+@RunWith(JUnit4.class)
+public class FormHandlerServletTest extends Mockito {
+
+    @InjectMocks
+    private FormHandlerServlet servlet = new FormHandlerServlet();
+
+    @Mock
+    HttpServletRequest request;
+
+    @Mock
+    HttpServletResponse response;
+
+    @Rule // JUnit 4 uses Rules for testing specific messages
+    public ExpectedException thrown = ExpectedException.none();
+
+    private final LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig()
+    .setDefaultHighRepJobPolicyUnappliedJobPercentage(100));
+
+    // keys
+    private static final String USER_FEED = "UserFeed";
+    private static final String PODCAST_TITLE = "title";
+    private static final String XML_STRING = "xmlString";
+    private static final String MP3_LINK = "mp3link"; // URL to existing MP3 file
+    private static final String TIMESTAMP = "timestamp";
+    private static final String EMAIL = "email"; 
+    private static final String ID = "id";
+
+    private static final String TEST_PODCAST_TITLE = "TEST_PODCAST_TITLE";
+    private static final String TEST_EMPTY_PODCAST_TITLE = "";
+    private static final String TEST_NULL_PODCAST_TITLE = null;
+    private static final String TEST_MP3_LINK = "TEST_MP3_LINK";
+    private static final String TEST_EMPTY_MP3_LINK = "";
+    private static final String TEST_NULL_MP3_LINK = null;
+    private static final long TEST_TIMESTAMP = System.currentTimeMillis();
+    private static final String TEST_ID = "123456";
+    private static final String TEST_ID_TWO = "789012";
+    private static final String TEST_XML_STRING;
+    private static final String TEST_PUBDATE = "2020/06/26 01:32:06";
+    private static final String TEST_EMAIL = "123@abc.com";
+    private static final String TEST_INCORRECT_EMAIL = "123@cde.com";
+    private static final String TEST_EMAIL_TWO = "456@abc.com";
+    private static final String EMPTY_STRING = "";
+    private static final String BASE_URL = "https://launchpod-step18-2020.appspot.com?id=";
+
+    @Before 
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        helper.setUp();
+    }
+
+    @After 
+    public void tearDown() {
+      helper.tearDown();
+    }
+
+    /**
+     * Creates a test user feed entity.
+    */
+    private Entity makeEntity(String title, String mp3Link, String xmlString) {
+        Entity userFeedEntity = new Entity(USER_FEED);
+        userFeedEntity.setProperty(PODCAST_TITLE, title);
+        userFeedEntity.setProperty(MP3_LINK, mp3Link);
+        userFeedEntity.setProperty(XML_STRING, xmlString);
+        return userFeedEntity;
+    }
+
+    /**
+     * Creates a test URL given the title of the podcast and the identifier key of its Datastore entity.
+    */
+    private String createUrl(String title, String id) {
```

Comment:
this is html, but it's not clear from the method name or comment that that's what it is. update the method name. i think you can remove the comment entirely if the method name and parameter names are obvious

Reply:
Makes sense. I actually am going to remove this method because I don't end up using it (based on recent changes)